src/test/java/com/solvd/eshop/AbstractTest.java

```diff
@@ -1,47 +1,59 @@
 package com.solvd.eshop;
 
 import com.solvd.eshop.utils.ConfigFileReader;
-import io.github.bonigarcia.wdm.WebDriverManager;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.apache.commons.io.FileUtils;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
 import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.chrome.ChromeDriver;
 import org.testng.ITestResult;
+import org.openqa.selenium.remote.RemoteWebDriver;
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.BeforeTest;
-import org.testng.reporters.FailedReporter;
+import org.testng.annotations.Parameters;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static com.solvd.eshop.utils.CapabilityFactory.getCapabilities;
 
 public abstract class AbstractTest {
     private static final Logger LOGGER = LogManager.getLogger(AbstractTest.class);
     private static final ThreadLocal<WebDriver> webDriver= new ThreadLocal<>();
 
-    @BeforeTest
-    public void beforeTestSetup() {
-        WebDriverManager.chromedriver().setup();
-    }
-
     @BeforeMethod
-    public void beforeMethodSetup() {
-        WebDriver driver = new ChromeDriver();
+    @Parameters({"browser"})
+    public void beforeMethodSetup(String browser) throws MalformedURLException {
+        WebDriver driver = new RemoteWebDriver(new URL(ConfigFileReader.getData("nodeUrl")), getCapabilities(browser));
+        driver.get(ConfigFileReader.getData("baseUrl"));
         LOGGER.info("21vek.by is opening");
-        driver.get(ConfigFileReader.getData("url"));
         webDriver.set(driver);
     }
 
+    public static void takeSnapShot(WebDriver webdriver, String fileWithPath) throws Exception {
+        DateFormat dateFormat = new SimpleDateFormat("yyyyMMdd HH:mm:ss");
+        Date date = new Date();
+        String strDate = dateFormat.format(date.getTime());
+        TakesScreenshot scrShot = ((TakesScreenshot) webdriver);
+        File SrcFile = scrShot.getScreenshotAs(OutputType.FILE);
+        File DestFile = new File(fileWithPath + strDate + ".png");
+        FileUtils.copyFile(SrcFile, DestFile);
+    }
+
     @AfterMethod(alwaysRun = true)
-    public void afterMethodSetup(ITestResult result) {
+    public void afterMethodSetup(ITestResult result) throws Exception {
         if (result.getStatus()==ITestResult.FAILURE) {
             LOGGER.error(result.getName()+ " is failed");
+            takeSnapShot(webDriver.get(), "./eshop/screenshots/");
```

Comment:
You can move **./eshop/screenshots/** to config file or constant class, bad practise to save paths in the code